src/wallet/rpcwallet.cpp

```diff
@@ -2779,11 +2779,15 @@ static RPCHelpMan upgradetohd()
 {
     return RPCHelpMan{"upgradetohd",
         "\nUpgrades non-HD wallets to HD.\n"
+        "\nIf your wallet is encrypted, the wallet passphrase must be supplied as it is used to encrypt the"
+        "\nHD chain. Unlocking the wallet beforehand has no bearing on the encryption process and cannot be"
+        "\nused in place of supplying the passphrase.\n"
```

Comment:
I don't think the explanation provided is useful to users; maybe could keep it as a code comment if you want; we should really just talk about requirements and observable behavior.

```suggestion
        "\nIf your wallet is encrypted, the wallet passphrase must be provided.\n"
```

Reply:
Resolved in latest push!